### PR TITLE
Rawread improvements

### PR DESCRIPTION
**Pull request by zagrodzki**
@kylelemons 
Actually read some data. Print out the collected data in a single-read mode.
Add a benchmark option that keeps reading and periodically reports the throughput.
Update iso code to use the length of the user-specified buffer as the basis for calculating number of iso transfers and for limiting the amount of returned data.

### CODE REVIEW

**Comment by zagrodzki** [comment]
Also note: letting the users manipulate the transfer size by controlling the provided buffer size allows them to control the tradeoff between latency and throughput. Low latency is more important in some applications, like audio processing.

**Comment by zagrodzki** [comment]
And also of interest, here are some benchmark results from my test runs:

with my "iso throughput experiments", transfers in flight set to 5, read size 61440 (20 iso packets, 3 * 1024 bytes each).

2017/02/08 23:54:50 Reading...
2017/02/08 23:54:54 23738108.65 B/s
2017/02/08 23:54:58 23752344.89 B/s
2017/02/08 23:55:02 23767290.78 B/s
2017/02/08 23:55:06 23752138.29 B/s
2017/02/08 23:55:10 23752216.49 B/s
2017/02/08 23:55:14 23751577.49 B/s

Same but with read size 6144 (2 iso packets):
2017/02/08 23:57:18 22605475.63 B/s
2017/02/08 23:57:22 22168941.04 B/s
2017/02/08 23:57:26 21614802.93 B/s
2017/02/08 23:57:30 21619392.52 B/s

Without the iso throughput experimental changes, size 92160 (30 iso packets):
2017/02/09 00:13:38 21482898.63 B/s
2017/02/09 00:13:42 21356483.88 B/s
2017/02/09 00:13:46 21025365.41 B/s
2017/02/09 00:13:50 20734278.85 B/s

And without the iso throughput changes, size 6144 (2 iso packets):
2017/02/09 00:15:23 9506157.03 B/s
2017/02/09 00:15:27 9085061.28 B/s
2017/02/09 00:15:31 9481184.93 B/s
2017/02/09 00:15:35 9782175.48 B/s

So clearly the gap between Submit and Wait is affecting the throughput.

File: rawread/main.go [hunk 1]
```diff
@@ -20,18 +20,26 @@ import (
...
 	"github.com/kylelemons/gousb/usbid"
 )
 
 var (
-	device   = flag.String("device", "vend:prod", "Device to which to connect")
-	config   = flag.Int("config", 1, "Endpoint to which to connect")
-	iface    = flag.Int("interface", 0, "Endpoint to which to connect")
-	setup    = flag.Int("setup", 0, "Endpoint to which to connect")
-	endpoint = flag.Int("endpoint", 1, "Endpoint to which to connect")
+	vid      = flag.Uint("vid", 0, "VID of the device to which to connect. Exclusive with bus/addr flags.")
```
Comment (kylelemons) [inline rawread/main.go]: consider making it --vidpid=vid:pid and using sscanf to make it somewhat clearer when reading the help page how to specify what you want

File: rawread/main.go [hunk 2]
```diff
@@ -20,18 +20,26 @@ import (
...
+	pid      = flag.Uint("pid", 0, "PID of the device to which to connect. Exclusive with bus/addr flags.")
+	bus      = flag.Uint("bus", 0, "Bus number for the device to which to connect. Exclusive with vid/pid flags.")
+	addr     = flag.Uint("addr", 0, "Address of the device to which to connect. Exclusive with vid/pid flags.")
+	config   = flag.Uint("config", 1, "Endpoint to which to connect")
+	iface    = flag.Uint("interface", 0, "Endpoint to which to connect")
+	setup    = flag.Uint("setup", 0, "Endpoint to which to connect")
+	endpoint = flag.Uint("endpoint", 1, "Endpoint to which to connect")
 	debug    = flag.Int("debug", 3, "Debug level for libusb")
+	size     = flag.Uint("read_size", 1024, "Number of bytes of data to read in a single transaction.")
+	bench    = flag.Bool("benchmark", false, "When true, keep reading from the endpoint and periodically report the measured throughput. If false,  only one read operation is performed and obtained data is printed to STDOUT.")
```
Comment (kylelemons) [inline rawread/main.go]: I think I would read and dump until ^C in the non-benchmark mode.

File: rawread/main.go (hunk 1 above)
Comment (zagrodzki) [inline rawread/main.go]: I went all the way and implemented separate parseVIDPID/parseBusAddr. Sscanf does not check for things like "is this an 8 bit unsigned number?" and it seemed easy to do good validation with strconv.

File: rawread/main.go (hunk 2 above)
Comment (zagrodzki) [inline rawread/main.go]: Done. With that I actually removed the benchmark mode - now it's easy to pipe the output into something like pipebench or cpipe.

**Pull request merged**